README
NAME
    TOBYINK::Acme::Acme::Acme - acme

SYNOPSIS
            use feature 'say';
            use TOBYINK::Acme::Acme::Acme;
        
            my $acme = 'TOBYINK::Acme::Acme::Acme'->new;
            say $acme->acme_acme;

DESCRIPTION
    This module is just a test. (Duh!)

  Methods
    `acme_acme`

BUGS
    Please report any bugs to
    <http://rt.cpan.org/Dist/Display.html?Queue=TOBYINK-Acme-Acme-Acme>.

SEE ALSO
    Acme.

AUTHOR
    Toby Inkster <tobyink@cpan.org>.

COPYRIGHT AND LICENCE
    This software is copyright (c) 2013 by Toby Inkster.

    This is free software; you can redistribute it and/or modify it under the
    same terms as the Perl 5 programming language system itself.

DISCLAIMER OF WARRANTIES
    THIS PACKAGE IS PROVIDED "AS IS" AND WITHOUT ANY EXPRESS OR IMPLIED
    WARRANTIES, INCLUDING, WITHOUT LIMITATION, THE IMPLIED WARRANTIES OF
    MERCHANTIBILITY AND FITNESS FOR A PARTICULAR PURPOSE.



Changes
TOBYINK-Acme-Acme-Acme
======================

Created:      2013-08-09
Home page:    <https://metacpan.org/release/TOBYINK-Acme-Acme-Acme>
Bug tracker:  <http://rt.cpan.org/Dist/Display.html?Queue=TOBYINK-Acme-Acme-Acme>
Maintainer:   Toby Inkster <mailto:tobyink@cpan.org>

0.002  2013-08-09

 - (Packaging) Include dependencies!

0.001  2013-08-09  Initial release




lib/TOBYINK/Acme/Acme/Acme.pm
package TOBYINK::Acme::Acme::Acme;

use strict;
use warnings;

BEGIN {
	$TOBYINK::Acme::Acme::Acme::AUTHORITY = 'cpan:TOBYINK';
	$TOBYINK::Acme::Acme::Acme::VERSION   = '0.002';
}

use Moo;

has acme_acme => (is => 'ro', default => 42);

1;

__END__

=pod

=encoding utf-8

=head1 NAME

TOBYINK::Acme::Acme::Acme - acme

=head1 SYNOPSIS

	use feature 'say';
	use TOBYINK::Acme::Acme::Acme;
	
	my $acme = 'TOBYINK::Acme::Acme::Acme'->new;
	say $acme->acme_acme;

=head1 DESCRIPTION

This module is just a test. (Duh!)

=head2 Methods

=over

=item C<acme_acme>

=back

=head1 BUGS

Please report any bugs to
L<http://rt.cpan.org/Dist/Display.html?Queue=TOBYINK-Acme-Acme-Acme>.

=head1 SEE ALSO

L<Acme>.

=head1 AUTHOR

Toby Inkster E<lt>tobyink@cpan.orgE<gt>.

=head1 COPYRIGHT AND LICENCE

This software is copyright (c) 2013 by Toby Inkster.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=head1 DISCLAIMER OF WARRANTIES

THIS PACKAGE IS PROVIDED "AS IS" AND WITHOUT ANY EXPRESS OR IMPLIED
WARRANTIES, INCLUDING, WITHOUT LIMITATION, THE IMPLIED WARRANTIES OF
MERCHANTIBILITY AND FITNESS FOR A PARTICULAR PURPOSE.



Makefile.PL
use strict;
use ExtUtils::MakeMaker 6.31;

my $meta = {
  "abstract"           => "acme",
  "author"             => ["Toby Inkster <tobyink\@cpan.org>"],
  "dynamic_config"     => 0,
  "generated_by"       => "Dist::Inkt::Profile::TOBYINK version 0.001, CPAN::Meta::Converter version 2.120921",
  "keywords"           => [],
  "license"            => ["perl_5"],
  "meta-spec"          => {
                            url => "http://search.cpan.org/perldoc?CPAN::Meta::Spec",
                            version => 2,
                          },
  "name"               => "TOBYINK-Acme-Acme-Acme",
  "no_index"           => { directory => ["inc", "t", "xt"] },
  "prereqs"            => {
                            configure => {
                              requires => { "ExtUtils::MakeMaker" => 6.31, "File::ShareDir::Install" => 0.02 },
                            },
                            runtime => { requires => { Moo => "1.000000" } },
                            test => {
                              recommends => { "File::ShareDir" => 0 },
                              requires   => { "Path::Tiny" => 0.027 },
                            },
                          },
  "provides"           => {
                            "TOBYINK::Acme::Acme::Acme" => { file => "lib/TOBYINK/Acme/Acme/Acme.pm", version => 0.002 },
                          },
  "release_status"     => "stable",
  "resources"          => {
                            bugtracker => {
                                            web => "http://rt.cpan.org/Dist/Display.html?Queue=TOBYINK-Acme-Acme-Acme",
                                          },
                            homepage   => "https://metacpan.org/release/TOBYINK-Acme-Acme-Acme",
                            license    => ["http://dev.perl.org/licenses/"],
                          },
  "version"            => 0.002,
  "x_contributors"     => [],
  "x_provides_scripts" => {},
};

my %dynamic_config;

my %WriteMakefileArgs = (
	ABSTRACT           => $meta->{abstract},
	AUTHOR             => $meta->{author},
	DISTNAME           => $meta->{name},
	DISTVNAME          => sprintf('%s-%s', $meta->{name}, $meta->{version}),
	EXE_FILES          => [ map $_->{file}, values %{ $meta->{x_provides_scripts} || {} } ],
	LICENSE            => $meta->{license}[0],
	NAME               => do { my $n = $meta->{name}; $n =~ s/-/::/g; $n },
	VERSION            => $meta->{version},
	%dynamic_config,
);

sub deps
{
	my %r;
	for my $stage (@_)
	{
		for my $dep (keys %{$meta->{prereqs}{$stage}{requires}})
		{
			my $ver = $meta->{prereqs}{$stage}{requires}{$dep};
			$r{$dep} = $ver if !exists($r{$dep}) || $ver >= $r{$dep};
		}
	}
	\%r;
}

my ($build_requires, $configure_requires, $runtime_requires, $test_requires);
if ('ExtUtils::MakeMaker'->VERSION >= 6.6303)
{
	$WriteMakefileArgs{BUILD_REQUIRES}     ||= deps('build');
	$WriteMakefileArgs{CONFIGURE_REQUIRES} ||= deps('configure');
	$WriteMakefileArgs{TEST_REQUIRES}      ||= deps('test');
	$WriteMakefileArgs{PREREQ_PM}          ||= deps('runtime');
}
elsif ('ExtUtils::MakeMaker'->VERSION >= 6.5503)
{
	$WriteMakefileArgs{BUILD_REQUIRES}     ||= deps('build');
	$WriteMakefileArgs{CONFIGURE_REQUIRES} ||= deps('configure', 'test');
	$WriteMakefileArgs{PREREQ_PM}          ||= deps('runtime');	
}
elsif ('ExtUtils::MakeMaker'->VERSION >= 6.52)
{
	$WriteMakefileArgs{CONFIGURE_REQUIRES} ||= deps('configure', 'build', 'test');
	$WriteMakefileArgs{PREREQ_PM}          ||= deps('runtime');	
}
else
{
	$WriteMakefileArgs{PREREQ_PM}          ||= deps('configure', 'build', 'test', 'runtime');	
}

{
	my $minperl = delete $WriteMakefileArgs{PREREQ_PM}{perl};
	exists($WriteMakefileArgs{$_}) && delete($WriteMakefileArgs{$_}{perl})
		for qw(BUILD_REQUIRES TEST_REQUIRES CONFIGURE_REQUIRES);
	if ($minperl and 'ExtUtils::MakeMaker'->VERSION >= 6.48)
	{
		$WriteMakefileArgs{MIN_PERL_VERSION} ||= $minperl;
	}
	elsif ($minperl)
	{
		die "Need Perl >= $minperl" unless $] >= $minperl;
	}
}

sub FixMakefile
{
	return unless -d 'inc';
	my $file = shift;
	
	local *MAKEFILE;
	open MAKEFILE, "< $file" or die "FixMakefile: Couldn't open $file: $!; bailing out";
	my $makefile = do { local $/; <MAKEFILE> };
	close MAKEFILE or die $!;
	
	$makefile =~ s/\b(test_harness\(\$\(TEST_VERBOSE\), )/$1'inc', /;
	$makefile =~ s/( -I\$\(INST_ARCHLIB\))/ -Iinc$1/g;
	$makefile =~ s/( "-I\$\(INST_LIB\)")/ "-Iinc"$1/g;
	$makefile =~ s/^(FULLPERL = .*)/$1 "-Iinc"/m;
	$makefile =~ s/^(PERL = .*)/$1 "-Iinc"/m;
	
	open  MAKEFILE, "> $file" or die "FixMakefile: Couldn't open $file: $!; bailing out";
	print MAKEFILE $makefile or die $!;
	close MAKEFILE or die $!;
}

use File::ShareDir::Install;
install_share 'share';
{ package MY; use File::ShareDir::Install qw(postamble) };

my $mm = WriteMakefile(%WriteMakefileArgs);
FixMakefile($mm->{FIRST_MAKEFILE} || 'Makefile');
exit(0);

